src/main/java/net/dv8tion/jda/core/requests/WebSocketClient.java

```diff
@@ -791,7 +641,12 @@ protected void invalidate()
         sessionId = null;
         sentAuthInfo = false;
 
-        chunkSyncQueue.clear();
+        locked("Interrupted while trying to invalidate queue", () -> {
+            chunkSyncQueue.clear();
+            ratelimitQueue.clear();
```

Comment:
We weren't clearing the ratelimit queue before.
Is there a specific reason we want to clear it now?

Reply:
The queue might contain lifecycle events that should not be sent again